BetterPasswordPlugin.inc.php

```diff
@@ -208,17 +236,62 @@ function checkPassword($hookName, $args) {
 			}
 		}
 	}
-
+	/**
+	 * Create a temporary file with the data of password blacklists.
+	 * @return string temporary filename string
+	 */
+         function handleTempFile() {
+            import('lib.pkp.classes.file.PrivateFileManager');
+            $fileMgr = new PrivateFileManager();
+            $siteDao = DAORegistry::getDAO('SiteDAO');
+            $site = $siteDao->getSite();
+            $minLengthPass = $site->getMinPasswordLength();
+            $tempFileDir = realpath($fileMgr->getBasePath()) . DIRECTORY_SEPARATOR . 'passTmp';
+            if (!$fileMgr->fileExists($tempFileDir, 'dir')) {
+			$success = $fileMgr->mkdirtree($tempFileDir);
+			if (!$success) {
+				// Files directory wrong configuration?
+				assert(false);
+				return false;
+			}
+		}
+            if(!file_exists($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile')) {
+                $fpTemp = fopen($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile', 'a');    
+                foreach ($this->getBlacklists() as $filename) {
+                    $fpPass = fopen($filename, "r");
+                    if (flock($fpTemp, LOCK_EX)) {
+                        while (!feof($fpPass)) {
+                            $passwordLine = fgets($fpPass);
+                            if (strlen($passwordLine) >= $minLengthPass) {
+                                fwrite($fpTemp, $passwordLine);
+                            }
+                        }
+			flock($fpTemp, LOCK_UN);
+                    } else {
+			// Couldn't lock the file.
+			return null;
+                    }
+                    fclose($fpPass);
+                }
+                fclose($fpTemp);
+                return ($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile');
+            }
+            else {
+                
+                return ($tempFileDir . DIRECTORY_SEPARATOR . 'tempPassFile');
+            }
+         }
+        
 	/**
 	 * Get the filename(s) of password blacklists.
 	 * @return array filename strings
 	 */
 	function getBlacklists() {
-		return array(
```

Comment:
Watch out for unintentional whitespace changes.

Reply:
There are still whitespace changes here.  We only want to see relevant changes within the pull request, so the whitespace changes here clutter the diff.